spec/models/user_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe "validations" do
+    USER_FIELDS = %i(email country).freeze
+    subject { FactoryGirl.build(:user) }
+
+    USER_FIELDS.each do |attr|
+      it { is_expected.to validate_presence_of attr }
+    end
+
+    it { is_expected.to validate_length_of :email }
```

Comment:
We can further refine this test by using the `is_at_least` and `is_at_most` methods to check min and max values. You can refer to the [Shoulda Matchers README](https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers) for all the different matchers and their options.